src/cargo/core/compiler/fingerprint.rs

```diff
@@ -1822,32 +1823,32 @@ fn find_stale_file(
                             }
                             hasher.input(&buffer[..count]);
                         }
-                        format!("{:?}", hasher.result())
+                        Some(format!("{:?}", hasher.result()))
                     }
                     FileHashAlgorithm::SvhInBin => {
                         debug!("found! got here");
-                        use object::Object;
-                        let v : Vec<u8> = vec![];
-                        let obj = object::read::File::parse(&v).unwrap();
-                        for sym in obj.symbols() {
-                            println!("{:#?}", sym);
+                        if path.ends_with(".rmeta") {
```

Comment:
```suggestion
                        if path.ends_with(".rlib") {
```

`.rmeta` is a bare metadata file like normally contained in an `.rlib`.

This by the way gives me the idea of putting the svh at a stable location in the crate metadata. executables, cdylibs and staticlibs could then include a metadata stub only containing the rustc version and svh. (the rustc version is already included in a stable location for better error reporting when mixing rustc versions)

Reply:
Look closer - here we're iterating over entries in that said `.rlib`. The svh is in the `lib.rmeta` dilename which is now `libcrate_name-svh.rmeta` in the ar - short of making it the first file in the archive I'm not sure we could do much less processing?

Reply:
`path` is the path of the `rlib` that contains `libcrate_name-svh.rmeta`, not `libcrate_name-svh.rmeta`.

Reply:
> short of making it the first file in the archive I'm not sure we could do much less processing?

Making it the first file will cause the linker to ignore all object files after it. Special care is taken to ensure the metadata file is the last one in the rlib.

Reply:
Ah my bad - was looking at the other location - good spot! Excellent point about having the metadata at the end.